src/coordination/include/coordination/coordinator_instance.hpp

```diff
@@ -161,6 +169,10 @@ class CoordinatorInstance {
   // Thread pool must be destructed first, because there is a possibility we are doing force reset in thread
   // while coordinator is destructed
   utils::ThreadPool thread_pool_{1};
+
+  CoordinatorInstanceManagementServer coordinator_management_server_;
+  utils::Synchronized<std::unordered_map<uint32_t, CoordinatorInstanceConnector>, utils::SpinLock>
```

Comment:
I think you would make your life easier with just std::vector + linear search since it is small collection. Map complicates things without much benefit but ok

Reply:
I need to think how to handle this better, I would leave it for the next PR to clean it up, I don't like it either